src/lsm/table_data_iterator.zig

```diff
@@ -104,7 +104,7 @@ pub fn TableDataIteratorType(comptime Storage: type) type {
                     .{ .from_local_or_global_storage = on_read },
                     &it.read,
                     it.context.addresses[index],
-                    it.context.checksums[index],
```

Comment:
It seems like this could be the right place to assert that the padding is zero? I don't think we assert this anywhere else?